plugins/zenoh-plugin-storage-manager/src/replica/aligner.rs

```diff
@@ -116,285 +121,226 @@ impl Aligner {
             for each_sub in diff_subintervals {
                 diff_string.push(each_sub.to_string());
             }
             let properties = format!(
                 "timestamp={}&{}=[{}]",
                 other.timestamp,
-                super::SUBINTERVALS,
+                SUBINTERVALS,
                 diff_string.join(",")
             );
             // expecting sample.value to be a vec of log entries with their checksum
-            let reply_content = self.perform_query(other_rep.to_string(), properties).await;
+            let (reply_content, mut no_err) = self.perform_query(other_rep, properties).await;
             let mut other_content: HashMap<u64, Vec<LogEntry>> = HashMap::new();
             for each in reply_content {
                 match serde_json::from_str(&each.value.to_string()) {
                     Ok((i, c)) => {
                         other_content.insert(i, c);
                     }
-                    Err(e) => error!("Error decoding reply: {}", e),
+                    Err(e) => {
+                        error!("[ALIGNER] Error decoding reply: {}", e);
+                        no_err = false;
+                    }
                 };
             }
             // get subintervals diff
-            return this.get_full_content_diff(other_content);
+            let result = this.get_full_content_diff(other_content);
+            (result, no_err)
+        } else {
+            (Vec::new(), true)
+        }
+    }
+
+    async fn perform_query(&self, from: &str, properties: String) -> (Vec<Sample>, bool) {
+        let mut no_err = true;
+        let selector = KeyExpr::from(&self.digest_key)
+            .join(&from)
+            .unwrap()
+            .with_parameters(&properties);
+        trace!("[ALIGNER] Sending Query '{}'...", selector);
+        let mut return_val = Vec::new();
+        if let Ok(replies) = self
+            .session
+            .get(&selector)
+            .consolidation(QueryConsolidation::AUTO)
+            .accept_replies(zenoh::query::ReplyKeyExpr::Any)
+            .res()
+            .await
+        {
+            while let Ok(reply) = replies.recv_async().await {
+                match reply.sample {
+                    Ok(sample) => {
+                        trace!(
+                            "[ALIGNER] Received ('{}': '{}')",
+                            sample.key_expr.as_str(),
+                            sample.value
+                        );
+                        return_val.push(sample);
+                    }
+                    Err(err) => {
+                        error!(
+                            "[ALIGNER] Received error for query on selector {} :{}",
+                            selector, err
+                        );
+                        no_err = false;
+                    }
+                }
+            }
+        } else {
+            error!("[ALIGNER] Query failed on selector `{}`", selector);
```

Comment:
Shouldn't `no_err` be set to `false` if the query fails ?
Also, it would help debugging to log the cause of failure. Use a `match` to log the error message.

Reply:
> Shouldn't no_err be set to false if the query fails ?
Thank you for catching this. It is in fact an error.

> Also, it would help debugging to log the cause of failure. Use a match to log the error message.
Agreed